Kitodo/src/main/java/org/kitodo/production/forms/createprocess/ProcessFieldedMetadata.java

```diff
@@ -705,6 +713,42 @@ public void preserve() throws InvalidMetadataValueException, NoSuchMetadataField
         }
     }
 
+    private void updateDivisionFromProcessDetail(String key, ProcessSimpleMetadata processDetail) throws InvalidMetadataValueException {
+        String simpleValue = extractSimpleValue(processDetail);
+        if (!processDetail.getSettings().isValid(simpleValue, getListForLeadingMetadataFields())) {
+            throw new InvalidMetadataValueException(key, simpleValue);
+        };
+        if (simpleValue == null) {
+            return;
+        }
+        updateDivision(key, simpleValue);
+    }
+
+    private String extractSimpleValue(ProcessDetail value) {
+        if (value instanceof ProcessTextMetadata) {
+            return ((ProcessTextMetadata) value).getValue();
+        } else if (value instanceof ProcessSelectMetadata) {
```

Comment:
In general, this refactoring is less object-oriented. It is not good practise to have lists of `if (x instanceof A) { … } else if (x instanceof B) { … } else if (x instanceof C) { … }` ...

Reply:
I tried to adress that remark by putting the necessary logic in the specific class and calling it using polymorphism.